ENH: Make MainWindow visible when calling PyDM without arguments.

hhslepicka opened the pull request:
It always bugged me the fact that we need to specify a file to be loaded.
This PR makes possible to just call `pydm` on the command line and get the Main Window to display so users can `Open File...` from the menu if they want.
It is a common mistake that happens mainly with new users but it is a nice "feature" to have.

teddyrendahl commented on pydm/main_window.py:
```diff
@@ -269,7 +272,12 @@ def toggle_status_bar(self, checked):
-        _, extension = path.splitext(self.current_file())
+        try:
+            curr_file = self.current_file()
+        except IndexError:
```
Can you add a `logger` message here? Uncomfortable with exception endpoints that are hard to track.

hhslepicka replied:
Addressed on the latest commit.

teddyrendahl commented on pydm/main_window.py:
```diff
@@ -327,6 +340,10 @@ def load_tool(self, checked):
     def reload_display(self, checked):
+        try:
+            curr_file = self.current_file()
+        except IndexError:
```
Log message here as well. This should probably be high level so that a user trying to press reload without a display button realizes what has happened.

hhslepicka replied:
Addressed on the latest commit.

teddyrendahl replied:
You sure it should just be `info`? Calling `reload_display` without a display seems like an `error`

hhslepicka replied:
I was interpreting it as a minor error, not something critical but your recommendation is valid so I addressed this at the latest commit and changed to `error`.

teddyrendahl commented on pydm/main_window.py:
```diff
@@ -304,7 +312,12 @@ def open_editor_generic(fname):
+        except IndexError:
+            curr_file = os.getcwd()
+
+        filename = QFileDialog.getOpenFileName(self, 'Open File...', os.path.dirname(curr_file), 'PyDM Display Files (*.ui *.py)')
```
I think this may go one level further up the tree than you think if there is no `current_file()`
```python
curr_file = os.getcwd()   # gets working directory
os.path.dirname(curr_file)  # gets directory of current working directory 
```

hhslepicka replied:
You are right!
Addressed on the latest commit.

hhslepicka commented:
@teddyrendahl I addressed the comments on the least commits.
I also added the logging configuration to the `pydm launcher` so we can really see the messages.
About returning `None` I will leave this to another opportunity as the whole machinery is set for the `IndexError`.
